pkg/controller/controllercmd/cmd.go

```diff
@@ -97,7 +96,7 @@ func hasServiceServingCerts(certDir string) bool {
 }
 
 // StartController runs the controller
-func (c *ControllerCommandConfig) StartController(stopCh <-chan struct{}) error {
+func (c *ControllerCommandConfig) StartController(stopCh chan struct{}) error {
```

Comment:
I don't like the change to `chan struct{}`. It makes the ownership unclear and leads to panic if both, caller and controller close the channel. Better use your own channel `exitOnChangeReactorCh` in addition locally here and combine it with `stopCh`. Pass the result to `Run`.

Reply:
Or even better: hide this extra channel from here by moving it into `staticPodOperatorControllers` and let `Run` stop if it is closed.